### ISSUE TITLE
Packages available for L2RPN 2022 competition

### ISSUE THREAD

**Issue by BDonnot** [rte-france/Grid2Op#332]
The complete list of packages available for 2022 competition is:

```
absl-py==1.1.0
acme==1.27.0
aiohttp==3.8.1
aiohttp-cors==0.7.0
aioredis==2.0.1
aiosignal==1.2.0
alabaster==0.7.12
ale-py==0.7.5
antlr4-python3-runtime==4.9.3
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
ase==3.22.1
astor==0.8.1
asttokens==2.0.5
astunparse==1.6.3
async-timeout==4.0.2
atari-py==0.2.9
attrs==21.4.0
autopep8==1.6.0
AutoROM==0.4.2
AutoROM.accept-rom-license==0.4.2
Babel==2.10.1
backcall==0.2.0
bayesian-optimization==1.2.0
BayesianOptimization==0.0.0
beautifulsoup4==4.11.1
bleach==5.0.0
blessings==1.7
branca==0.5.0
cachetools==5.2.0
certifi==2022.5.18.1
cffi==1.15.0
cftime==1.6.0
charset-normalizer==2.0.12
chex==0.1.3
Chronix2Grid==1.1.0.post1
click==8.1.3
cloudpickle==2.1.0
cma==3.2.2
colorama==0.4.4
colorful==0.5.4
colorlover==0.3.0
cryptography==37.0.2
cufflinks==0.17.3
cvxopt==1.3.0
cvxpy==1.2.1
cycler==0.11.0
cylp==0.91.5
debugpy==1.6.0
decorator==5.1.1
defusedxml==0.7.1
Deprecated==1.2.13
deprecation==2.1.0
dgl==0.8.2
dglgo==0.0.1
distlib==0.3.4
dm-acme==0.4.0
dm-env==1.5
dm-launchpad==0.5.0
dm-sonnet==2.0.0
dm-tree==0.1.7
docutils==0.18.1
dopamine-rl==4.0.5
ecos==2.0.10
entrypoints==0.4
et-xmlfile==1.1.0
executing==0.8.3
fastjsonschema==2.15.3
filelock==3.7.1
flatbuffers==1.12
flax==0.3.4
folium==0.12.1.post1
fonttools==4.33.3
frozenlist==1.3.0
gast==0.4.0
gin-config==0.5.0
google-auth==2.6.6
google-auth-oauthlib==0.4.6
google-pasta==0.2.0
Grid2Op==1.7.2
grpcio==1.43.0
gym==0.21.0
h5py==3.7.0
hydra-core==1.2.0
idna==3.3
imageio==2.19.3
imagesize==1.3.0
importlib-metadata==4.11.4
importlib-resources==5.7.1
iniconfig==1.1.1
ipykernel==6.13.1
ipython==8.4.0
ipython-genutils==0.2.0
ipywidgets==7.7.0
isodate==0.6.1
isort==5.10.1
jax==0.3.13
jaxlib==0.3.10
jedi==0.18.1
Jinja2==3.1.2
joblib==1.1.0
josepy==1.13.0
jraph==0.0.5.dev0
json5==0.9.8
jsonschema==4.6.0
jupyter==1.0.0
jupyter-client==7.3.3
jupyter-console==6.4.3
jupyter-core==4.10.0
jupyterlab-pygments==0.2.2
jupyterlab-widgets==1.1.0
keras==2.9.0
Keras-Preprocessing==1.1.2
kiwisolver==1.4.2
l2rpn-baselines==0.6.0.post1
libclang==14.0.1
LightSim2Grid==0.7.0.post1
llvmlite==0.38.1
lxml==4.9.0
lz4==4.0.1
Markdown==3.3.7
MarkupSafe==2.1.1
matplotlib==3.5.2
matplotlib-inline==0.1.3
mirakuru==2.4.2
mistune==0.8.4
mock==4.0.3
msgpack==1.0.4
multidict==6.0.2
nbclient==0.6.4
nbconvert==6.5.0
nbformat==5.4.0
nest-asyncio==1.5.5
netCDF4==1.5.8
networkx==2.8.3
nevergrad==0.5.0
nose==1.3.7
notebook==6.4.11
numba==0.55.2
numexpr==2.8.1
numpy==1.22.4
numpydoc==1.3.1
oauthlib==3.2.0
omegaconf==2.2.2
opencv-python==4.5.5.64
openpyxl==3.0.10
opt-einsum==3.3.0
optax==0.1.2
ortools==9.3.10497
osqp==0.6.2.post5
packaging==21.3
paddle-bfloat==0.1.2
paddlepaddle==2.3.0
pandapower==2.9.0
pandas==1.4.2
pandocfilters==1.5.0
parso==0.8.3
patsy==0.5.2
pexpect==4.8.0
pickleshare==0.7.5
Pillow==9.1.1
platformdirs==2.5.2
plotly==5.8.0
pluggy==0.13.1
ply==3.11
port-for==0.6.2
portpicker==1.5.0
prometheus-client==0.14.1
prompt-toolkit==3.0.29
protobuf==3.19.4
psutil==5.9.1
ptyprocess==0.7.0
pure-eval==0.2.2
py==1.11.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pybind11==2.9.2
pycodestyle==2.8.0
pycparser==2.21
pydantic==1.9.1
pygame==2.0.1
pygifsicle==1.0.5
pyglet==1.5.26
Pygments==2.12.0
Pympler==1.0.1
Pyomo==6.4.1
pyOpenSSL==22.0.0
pyparsing==3.0.9
pypsa==0.19.3
pyRFC3339==1.1
pyrsistent==0.18.1
pytest==6.2.4
pytest-redis==2.1.1
pytest-timeout==1.4.2
pytest-tornasync==0.6.0.post2
python-dateutil==2.8.2
pytz==2022.1
PyUtilib==6.0.0
PyWavelets==1.3.0
PyYAML==6.0
pyzmq==23.1.0
qdldl==0.1.5.post2
qtconsole==5.3.1
QtPy==2.1.0
ray==1.12.1
redis==4.3.3
requests==2.27.1
requests-oauthlib==1.3.1
requests-toolbelt==0.9.1
rsa==4.8
ruamel.yaml==0.17.21
ruamel.yaml.clib==0.2.6
scikit-image==0.19.2
scikit-learn==1.1.1
scipy==1.8.1
scs==3.2.0
seaborn==0.11.1
Send2Trash==1.8.0
six==1.16.0
sklearn==0.0
snowballstemmer==2.2.0
soupsieve==2.3.2.post1
spektral==1.1.0
Sphinx==5.0.1
sphinxcontrib-applehelp==1.0.2
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==2.0.0
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.5
stable-baselines3==1.5.0
stack-data==0.2.0
statsmodels==0.13.2
tables==3.7.0
tabulate==0.8.9
tenacity==8.0.1
tensorboard==2.9.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
tensorboardX==2.5.1
tensorflow==2.9.1
tensorflow-addons==0.17.0
tensorflow-estimator==2.9.0
tensorflow-io-gcs-filesystem==0.26.0
tensorflow-probability==0.16.0
termcolor==1.1.0
terminado==0.15.0
tf-slim==1.1.0
threadpoolctl==3.1.0
tianshou==0.4.8
tifffile==2022.5.4
tinycss2==1.1.1
toml==0.10.2
tomli==2.0.1
toolz==0.11.2
torch==1.11.0+cpu
torch-cluster==1.6.0
torch-geometric==2.0.4
torch-scatter==2.0.9
torch-sparse==0.6.13
torch-spline-conv==1.2.1
torchaudio==0.11.0+cpu
torchvision==0.12.0+cpu
tornado==6.1
tqdm==4.64.0
traitlets==5.2.2.post1
typeguard==2.13.3
typer==0.4.1
typing_extensions==4.2.0
urllib3==1.26.9
virtualenv==20.14.1
wcwidth==0.2.5
webencodings==0.5.1
Werkzeug==2.1.2
widgetsnbextension==3.6.0
wrapt==1.14.1
xarray==2022.3.0
xgboost==1.6.1
XlsxWriter==3.0.3
yarl==1.7.2
zipp==3.8.0
```

Want to use a package not on the list ? Write down its name and we'll see what we can do
 **NB** packages can only be added BEFORE the end of the warmup phase so BEFORE the validation starts. Once the validation phase starts, no more packages (unless if a critical bug is found) will be modified.
 
 # EDIT 2022/07/05

Upgrade of some package versions:

- grid2op ~~1.7.1~~ -> 1.7.2
- lightsim2grid ~~0.7.0~~ -> 0.7.0.post1
- l2rpn-baselines ~~0.6.0~~ -> 0.6.0.post1

**Comment by BDonnot** [comment]
**NB** This list of packages is only usefull during the test phase on codalab. You are free to use ANY packages you want to train / fine tune / improve your agent locally.

**Comment by BDonnot** [comment]
The file to replicate the environment used to evaluate the submission is:

[Dockerfile.zip](https://github.com/rte-france/Grid2Op/files/8876811/Dockerfile.zip)

**Comment by Afan98412** [comment]
ninhao,wo xiang wenyixia, zhege bisai bixu zai unbuntu shang me ?

**Comment by BDonnot** [comment]
Hello,

I am not sure I understand your comment.

But the warm-up phase is over. It's not possible to add packages right now (to be fair to participants that already submitted agents)

Let me know if I can be of further.m assistance

**Comment by Afan98412** [comment]
您好，我想跑一下这个代码，但是没有跑通，可以加您微信咨询一下么？
13290551696




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "rte-france/Grid2Op"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年7月22日(星期五) 晚上7:29
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [rte-france/Grid2Op] Packages available for L2RPN 2022 competition (Issue #332)





 
Hello,
 
I am not sure I understand your comment.
 
But the warm-up phase is over. It's not possible to add packages right now (to be fair to participants that already submitted agents)
 
Let me know if I can be of further.m assistance

**Comment by BDonnot** [comment]
I used Google translate to translate your question.

Can you join our discord?

https://discord.gg/cYsYrPT

 You will receive technical help there. I don't have wechat.

To replicate the basis of the environment of the competition you can do:


```
pip install "grid2op==1.7.2" "lightsim2grid==0.7.0.post1" "l2rpn_baselines==0.6.0.post1"
```

(with a working python installation on your computer)

**Comment by Afan98412** [comment]
OKOK,thank you vary much


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "rte-france/Grid2Op"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年7月22日(星期五) 晚上8:01
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [rte-france/Grid2Op] Packages available for L2RPN 2022 competition (Issue #332)





 
I used Google translate to translate your question.
 
Can you join our discord?
 
https://discord.gg/cYsYrPT
 
You will receive technical help there. I don't have wechat.
 
To replicate the basis of the environment of the competition you can do:
 pip install "grid2op==1.7.2" "lightsim2grid==0.7.0.post1" "l2rpn_baselines==0.6.0.post1"  
(with a working python installation on your computer)
